packages/swing-store/src/snapStore.js

```diff
@@ -186,15 +250,51 @@ export function makeSnapStore(
       autoClose: false,
     });
     try {
+      // This nested await is safe because "synchronous-throw-impossible".
+      // To help ensure this, we changed `fsStreamReady` to be async.
+      // We believe (but someone should check) that `Promise.all` is
+      // already "synchronous-throw-impossible".
+      // eslint-disable-next-line @jessie.js/no-nested-await
       await Promise.all([
         fsStreamReady(sourceStream),
         fsStreamReady(destinationStream),
       ]);
+      // ASSUMPTION: functions returned by `promisify` are
+      // "synchronous-throw-impossible", such as an async function.
+      //
+      // If so, this nested await is safe because
+      // "synchronous-throw-impossible", because `pipe` is locally defined
+      // as the result of `promisify`.
+      //
+      // For reference, the current implementation of `promisify` is at
+      // https://github.com/nodejs/node/blob/v16.17.0/lib/internal/util.js#L324-L375
+      // for which this guarantee is currently likely true, even though the
+      // returned function is not declared `async`. If we're worried that
+      // this guarantee might be lost under maintenance, we can make the
+      // `pipe` call below into an "IIAFE".
+      //
+      // eslint-disable-next-line @jessie.js/no-nested-await
       await pipe(sourceStream, transform, destinationStream);
       if (flush) {
-        await destination.sync();
+        // This nested await has been made SAFER by "IIAFE". Note that
+        // doing so does *not* postpone when the `sync()` happens,
+        // which would be especially scary for something named `sync()`.
+        // Rather, it potentially postpones when the `close()`s happen
+        // in case `sync()` throws. With the IIAFE, if this line is
+        // executed, then the `close()`s would always happen in a later turn.
+        //
+        // But this nested await is not actually safe because it is nested in
+        // an `if`, the other branch of which proceeds synchrounously. The
+        // `if` is not terminal because of the stateful `close()`s afterward.
+        // On the else branch, those `close()`s would happen synchronously.
+        // eslint-disable-next-line @jessie.js/no-nested-await
+        await (async () => destination.sync())();
```

Comment:
As above, in this case it happens to be safe to introduce a new interleaving point before the `sync()` call, because the immediately preceding stateful code is another await. But it is not a technique that works in general, so we should be careful about making it a habit.